### ISSUE TITLE
OP_BLOCK critical bug

### ISSUE THREAD

**Issue by supahero1** [Qwokka/WAIL#8]
Hey,
I've just noticed that OP_BLOCK in wail reads a single byte ahead, which is supposed to be the block's return type, but in fact it should be a varuint read, because whatever type follows OP_BLOCK is allowed to be one of the module type definitions. I was just working on a WASM module that had a hand-crafted block like this:
```wat
    block  ;; label = @1
      f32.const 0x1.fffe1cp-96 (;=2.52432e-29;)
      drop
    end)
```
That translates to `[2, 128, 4, 67, 14, 255, 255, 15, 26, 11, 11]`. As you can see, once WAIL reads 128, it then thinks the next opcode is 4 (OP_IF), once again it reads one byte, then the opcode it encounters is 14 (OP_BR_TABLE), and then it reads the varuint value of 245888 or so. But of course that isn't the end of problems, because then it reads 245888 more varuints, which all end up outside of the memory bounds so it keeps growing the memory and reading zeroes constantly. All this is handled by the browser perfectly, since the browser reads a varuint. It would be cool to see WAIL fix this bug to read a varuint too :)
I would assume that OP_LOOP and OP_IF have the same vulnerability.

**Comment by Qwokka** [comment]
Hey @supahero1 

Thanks for catching this! As I understand it, these 3 instructions each used to just be followed by a fixed varuint7 but that's no longer the case. This should be fixed with df65e035fa5bb1ea62866dab654653e488a79db3

Thanks again!
Jack

**Comment by Nul-led** [comment]
expected this to take longer lol
